python/feldera/pipeline.py

```diff
@@ -344,3 +344,39 @@ def get(name: str, client: FelderaClient) -> 'Pipeline':
         except FelderaAPIError as err:
             if err.status_code == 404:
                 raise RuntimeError(f"Pipeline with name {name} not found")
+
+    def query(self, query: str, fmt: QueryResponseFormat) -> Generator[Mapping[str, Any], None, None]:
+        """
+        Executes an ad-hoc SQL query on this pipeline and returns the result in the specified format.
+
+        :param query: The SQL query to be executed.
+        :param fmt: An instance of the :class:`.QueryResponseFormat` enum, specifying the output format:
+
+            - `QueryResponseFormat.JSON`: Returns a generator that yields rows as Python dictionaries.
+            - `QueryResponseFormat.PARQUET`: Returns the result as a binary blob in Parquet format.
+
+        :return: A generator that yields the rows of the result as Python dictionaries when the format is JSON,
+                 or binary data when the format is Parquet.
+        """
+
+        if self.status() not in [
+            PipelineStatus.RUNNING,
+            PipelineStatus.PAUSED,
+        ]:
+            raise RuntimeError("Pipeline must be running or paused to run a query")
+        return self.client.query(self.name, query, fmt.name)
```

Comment:
For Parquet this function returns bytes, not `Generator`, so this doesn't seem correct?